src/main/groovy/com/codebrig/arthur/observe/structure/filter/operator/misc/TernaryOperatorFilter.groovy

```diff
@@ -25,14 +26,15 @@ class TernaryOperatorFilter extends StructureFilter<TernaryOperatorFilter, Void>
                         new RoleFilter("RIGHT")
                 ),
                 MultiFilter.matchAll(
-                        new RoleFilter("CONDITION"), new RoleFilter("EXPRESSION")
+                        new RoleFilter("CONDITION"), new RoleFilter("EXPRESSION"),
+                        new InternalRoleFilter().reject("Prop_InitOperand2")
                 )
         )
     }
 
     @Override
     boolean evaluate(SourceNode node) {
-        if (node.internalType == "ConditionalExpression") {
+        if (Objects.requireNonNull(node).internalType == "ConditionalExpression") {
```

Comment:
You can't require non-null here. `node` is allowed to be null.

Reply:
@BFergerson Please see my latest commit https://github.com/CodeBrig/Arthur/pull/86/commits/058027cc22beaf457f932ab399abe9aced5e5a52.